worlds/metroidprime/data/ChozoRuins.py

```diff
@@ -0,0 +1,996 @@
+from BaseClasses import CollectionState
+from ..BlastShieldRando import BlastShieldType
+from ..DoorRando import DoorLockType
+from ..LogicCombat import can_combat_flaahgra, can_combat_ghosts
+from ..Items import SuitUpgrade
+from ..data.AreaNames import MetroidPrimeArea
+from .RoomData import AreaData, PickupData, RoomData
+from .DoorData import DoorData
+from ..Logic import (
+    can_bomb,
+    can_boost,
+    can_grapple,
+    can_heat,
+    can_ice_beam,
+    can_missile,
+    can_morph_ball,
+    can_move_underwater,
+    can_plasma_beam,
+    can_power_beam,
+    can_power_bomb,
+    can_scan,
+    can_space_jump,
+    can_spider,
+    can_super_missile,
+    can_wave_beam,
+    has_energy_tanks,
+    has_power_bomb_count,
+)
+from ..data.Tricks import Tricks
+from .RoomNames import RoomName
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from .. import MetroidPrimeWorld
+
+
+def can_exit_ruined_shrine(world: "MetroidPrimeWorld", state: CollectionState) -> bool:
+    item1 = world.get_location(
+        "Chozo Ruins: Ruined Shrine - Plated Beetle"
+    ).item
+    if can_morph_ball(world, state) or can_space_jump(world, state):
+        return True
+    if (
+        item1 is not None
+        and item1.name in ["Morph Ball", "Space Jump Boots"]
+        and item1.player == world.player
+    ):
+        return True
+    return False
+
+
+def can_climb_sun_tower(world: "MetroidPrimeWorld", state: CollectionState) -> bool:
+    return (
+        can_spider(world, state)
+        and can_super_missile(world, state)
+        and can_bomb(world, state)
+    )
+
+
+def can_flaahgra(world: "MetroidPrimeWorld", state: CollectionState) -> bool:
```

Comment:
I'm probably missing something here, but if memory serves only Power Beam or Missiles can knock down the sun reflectors, and you need to do that for the first time to unlock the bomb slots even if you dodge the tentacles for the other phases. As best as I can tell, there's no protection against starting with Ice/Wave/Plasma and bombs and no missiles in Sunchamber Lobby, which would render Flaahgra unbeatable.

Reply:
Ha nice catch on this, I just tested and you're absolutely right